tests/server/test_params.py

```diff
@@ -267,35 +267,38 @@ def test_parse_day_value(self):
     def test_parse_time_arg(self):
         with self.subTest("empty"):
             with app.test_request_context("/"):
-                self.assertEqual(parse_time_arg(), [])
+                self.assertEqual(parse_time_arg(), None)
         with self.subTest("single"):
             with app.test_request_context("/?time=day:*"):
-                self.assertEqual(parse_time_arg(), [TimePair("day", True)])
+                self.assertEqual(parse_time_arg(), TimePair("day", True))
             with app.test_request_context("/?time=day:20201201"):
-                self.assertEqual(parse_time_arg(), [TimePair("day", [20201201])])
+                self.assertEqual(parse_time_arg(), TimePair("day", [20201201]))
         with self.subTest("single list"):
             with app.test_request_context("/?time=day:20201201,20201202"):
-                self.assertEqual(parse_time_arg(), [TimePair("day", [20201201, 20201202])])
+                self.assertEqual(parse_time_arg(), TimePair("day", [20201201, 20201202]))
         with self.subTest("single range"):
             with app.test_request_context("/?time=day:20201201-20201204"):
-                self.assertEqual(parse_time_arg(), [TimePair("day", [(20201201, 20201204)])])
+                self.assertEqual(parse_time_arg(), TimePair("day", [(20201201, 20201204)]))
         with self.subTest("multi"):
-            with app.test_request_context("/?time=day:*;week:*"):
-                self.assertEqual(parse_time_arg(), [TimePair("day", True), TimePair("week", True)])
-            with app.test_request_context("/?time=day:20201201;week:202012"):
+            with app.test_request_context("/?time=day:*;day:20201201"):
                 self.assertEqual(
                     parse_time_arg(),
-                    [TimePair("day", [20201201]), TimePair("week", [202012])],
+                    TimePair("day", True)
                 )
-        with self.subTest("hybrid"):
-            with app.test_request_context("/?time=day:*;day:20202012;week:202101-202104"):
+            with app.test_request_context("/?time=week:*;week:202012"):
                 self.assertEqual(
                     parse_time_arg(),
-                    [
-                        TimePair("day", True),
-                        TimePair("day", [20202012]),
-                        TimePair("week", [(202101, 202104)]),
-                    ],
+                    TimePair("week", True)
+                )
```

Comment:
Try testing the mixed time type exception here:

```py
        with self.subTest("hybrid"):
            with app.test_request_context("/?time=day:*;day:20202012;week:202101-202104"):
                with pytest.raises(ValidationFailedException(match=r"*mixes \"day\" and \"week\" time types.")):
                    parse_time_arg()
```

Reply:
(Did this with `assertRaisesRegex` which seems identical!)